disnake/types/channel.py

```diff
@@ -138,7 +140,8 @@ class ThreadChannel(_BaseChannel, _ThreadChannelOptional):
     nsfw: bool
     last_message_id: Optional[Snowflake]
     rate_limit_per_user: int
-    message_count: int
+    message_count: Optional[int]
+    total_message_sent: Optional[int]
```

Comment:
@shiftinv are we sure these are optional?

Reply:
The other thread model type assumes they are always set, which may be a valid assumption but I'd rather not take any chances.
The definitions in both types should be in their respective `_Optional` parents, and perhaps default to `0` when parsing?

Reply:
Just noticed that at least in this case they're already part of `_ThreadChannelOptional`, so they should be removed from `ThreadChannel` anyway